lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -106,20 +110,25 @@ import qualified Data.List.NonEmpty as NE
 data NetworkLayer m block = NetworkLayer
     { chainSync
         :: Tracer IO ChainFollowLog
-        -> ChainFollower m
-            ChainPoint
-            BlockHeader
-            block
+        -> ChainFollower m ChainPoint BlockHeader (NonEmpty block)
```

Comment:
why adding `NonEmpty block` rather than `block` ?

Reply:
In order to use `ChainFollower` with `lightSync` as well, I had to change its argument from the "singular" `block` to the "plural" `blocks`. In other words, the type argument of `ChainFollower` now refers to a collection of blocks rather than a single block.